bundles/binding/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/SerialDevice.java

```diff
@@ -280,39 +280,35 @@ public void serialEvent(SerialPortEvent event) {
                     if (eventPublisher != null) {
                         if (configMap != null && !configMap.isEmpty()) {
                             for (Entry<String, ItemType> entry : configMap.entrySet()) {
-
+                                String pattern = entry.getValue().pattern;
                                 // use pattern
-                                if (entry.getValue().pattern != null) {
-
-                                    if (transformationService == null) {
-                                        logger.error("No transformation service available!");
+                                if (pattern != null) {
+                                    try {
+                                        String[] matches = RegexPatternMatcher.getMatches(pattern, result);
 
-                                    } else {
-                                        try {
-                                            String value = transformationService.transform(entry.getValue().pattern,
-                                                    result);
+                                        for (int i = 0; i < matches.length; i++) {
+                                            String match = matches[i];
 
                                             try {
                                                 State state = null;
 
                                                 if (entry.getValue().type.equals(NumberItem.class)) {
-                                                    state = new DecimalType(value);
+                                                    state = new DecimalType(match);
                                                 } else if (entry.getValue().type == RollershutterItem.class) {
-                                                    state = new PercentType(value);
+                                                    state = new PercentType(match);
                                                 } else {
-                                                    state = new StringType(value);
+                                                    state = new StringType(match);
                                                 }
 
                                                 eventPublisher.postUpdate(entry.getKey(), state);
                                             } catch (NumberFormatException e) {
                                                 logger.warn("Unable to convert regex result '{}' for item {} to number",
                                                         new String[] { result, entry.getKey() });
                                             }
-                                        } catch (TransformationException e) {
-                                            logger.error("Unable to transform!", e);
                                         }
+                                    } catch (TransformationException e) {
+                                        logger.error("Unable to transform!", e);
```

Comment:
Not shure, TransformationException is thrown only if you provide null pattern or result